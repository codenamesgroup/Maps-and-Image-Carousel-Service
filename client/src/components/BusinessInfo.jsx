import React from 'react';
import PropTypes from 'prop-types';

import styled from 'styled-components';
import * as FontAwesome from 'react-icons/lib/fa';

import { Map } from './Map.jsx';

class BusinessInfo extends React.PureComponent {
	static propTypes = {
		business: PropTypes.object,
	};

	static defaultProps = {
		business: {},
	};

	static container = styled.div`
		border: 1px solid #cccccc;
		margin: 0;
		padding: 4px;
	`;

	static infoContainer = styled.div`
		display: grid;
		grid-template-columns: 32px 1fr;
		padding: 8px;
	`;

	static textItem = styled.p`
		fontSize: 10pt;
		margin: auto 0;
		padding: 0;
	`;

	static link = BusinessInfo.textItem.extend`
		color: blue;
		cursor: pointer;
	`;

	static address = BusinessInfo.textItem.extend`
		font-weight: bold;
	`;

	render() {
		const iconStyle = {
			width:'18px',
			height:'18px',
			margin: '4px auto',
			color: 'darkgray',
		}

		return (
			<BusinessInfo.container>
				<Map longitude={this.props.business.longitude || 0.0} latitude={this.props.business.latitude || 0.0}/>

				<BusinessInfo.infoContainer>

					<FontAwesome.FaMapMarker style={iconStyle}/>
					<div>
						<BusinessInfo.address>{`${this.props.business.address}`}</BusinessInfo.address>
						<BusinessInfo.address>{
							`${this.props.business.city}, ${this.props.business.state} ${this.props.business.postal_code}`
						}</BusinessInfo.address>
					</div>

					<FontAwesome.FaMapSigns style={iconStyle}/>
					<BusinessInfo.link>Get Directions</BusinessInfo.link>

					 {/* no phone numbers in the database */}
					<FontAwesome.FaPhone style={iconStyle}/>
					<BusinessInfo.textItem>(545) 523-7964</BusinessInfo.textItem>

					<FontAwesome.FaExternalLink style={iconStyle}/>
					<BusinessInfo.link>{`${this.props.business.name}.com`}</BusinessInfo.link>

					<FontAwesome.FaMobile style={iconStyle}/>
					<BusinessInfo.link>Send to your phone</BusinessInfo.link>

				</BusinessInfo.infoContainer>
			</BusinessInfo.container>
		);
	}
};

export { BusinessInfo };