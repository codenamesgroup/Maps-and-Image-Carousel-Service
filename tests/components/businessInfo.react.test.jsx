import React from 'react';
import { shallowToJson } from 'enzyme-to-json';

import { BusinessInfo } from './../../client/src/components/BusinessInfo.jsx';

describe('<BusinessInfo/>', () => {
	let component = shallow(<BusinessInfo business={{}}/>);

	it('should render with empty business object', () => {
		expect(shallowToJson(component)).toMatchSnapshot();
	});
});