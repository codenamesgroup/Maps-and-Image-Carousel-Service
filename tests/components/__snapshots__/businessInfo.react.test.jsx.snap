// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusinessInfo/> should render with empty business object 1`] = `
<styled.div>
  <Map
    latitude={0}
    longitude={0}
  />
  <styled.div>
    <FaMapMarker
      style={
        Object {
          "color": "darkgray",
          "height": "18px",
          "margin": "4px auto",
          "width": "18px",
        }
      }
    />
    <div>
      <styled.p>
        undefined
      </styled.p>
      <styled.p>
        undefined, undefined undefined
      </styled.p>
    </div>
    <FaMapSigns
      style={
        Object {
          "color": "darkgray",
          "height": "18px",
          "margin": "4px auto",
          "width": "18px",
        }
      }
    />
    <styled.p>
      Get Directions
    </styled.p>
    <FaPhone
      style={
        Object {
          "color": "darkgray",
          "height": "18px",
          "margin": "4px auto",
          "width": "18px",
        }
      }
    />
    <styled.p>
      (545) 523-7964
    </styled.p>
    <FaExternalLink
      style={
        Object {
          "color": "darkgray",
          "height": "18px",
          "margin": "4px auto",
          "width": "18px",
        }
      }
    />
    <styled.p>
      undefined.com
    </styled.p>
    <FaMobile
      style={
        Object {
          "color": "darkgray",
          "height": "18px",
          "margin": "4px auto",
          "width": "18px",
        }
      }
    />
    <styled.p>
      Send to your phone
    </styled.p>
  </styled.div>
</styled.div>
`;
